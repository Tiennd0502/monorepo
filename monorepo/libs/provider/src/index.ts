export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
