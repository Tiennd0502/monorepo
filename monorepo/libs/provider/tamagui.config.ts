import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
