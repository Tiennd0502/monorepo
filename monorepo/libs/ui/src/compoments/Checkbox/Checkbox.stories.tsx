import { Meta, StoryObj } from '@storybook/react';

import Checkbox from '.';

const meta: Meta<typeof Checkbox> = {
  title: 'Components/Checkbox',
  component: Checkbox,
};

export default meta;

type Story = StoryObj<typeof Checkbox>;

export const Default: Story = {
  args: {
    label: 'Label checkbox',
  },
};

export const Checked: Story = {
  args: {
    label: 'Label checkbox',
    defaultChecked: true,
  },
};
