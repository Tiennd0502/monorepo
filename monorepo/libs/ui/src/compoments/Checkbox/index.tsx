import {
  Label,
  XStack,
  styled,
  CheckboxProps as TCheckboxProps,
  Checkbox as TCheckbox,
} from 'tamagui';
import { useId, memo } from 'react';

import { CheckIcon } from '../icons';

const CheckboxFrame = styled(TCheckbox, {
  borderWidth: 1,
  borderColor: '$borderColor',
  borderRadius: '$small',
  alignItems: 'center',
  justifyContent: 'center',

  variants: {
    checked: {
      indeterminate: {},
      true: {
        backgroundColor: '$primary',
      },
      false: {
        backgroundColor: '$secondary',
      },
    },

    size: {
      md: {
        width: '$5',
        height: '$5',
      },
    },
  } as const,

  defaultVariants: {
    checked: false,
    size: 'md',
  },
});

const StyledLabel = styled(Label, {
  variants: {
    checked: {
      indeterminate: {},
      true: {
        color: '$primary',
      },
      false: {
        color: '$textDefault',
      },
    },

    size: {
      md: {
        fontSize: '$4.5',
      },
    },
  } as const,

  defaultVariants: {
    size: 'md',
  },
});

interface CheckboxProps extends Omit<TCheckboxProps, 'size'> {
  label?: string;
  size?: 'md';
}

const Checkbox = ({
  size = 'md',
  label = '',
  checked = false,
  ...rest
}: CheckboxProps) => {
  const id = useId();

  return (
    <XStack width={300} alignItems="center" gap="$2">
      <CheckboxFrame id={id} size={size} checked={checked} {...rest}>
        <TCheckbox.Indicator>
          <CheckIcon />
        </TCheckbox.Indicator>
      </CheckboxFrame>

      {label && (
        <StyledLabel checked={checked} size={size} htmlFor={id}>
          {label}
        </StyledLabel>
      )}
    </XStack>
  );
};

export default memo(Checkbox);
