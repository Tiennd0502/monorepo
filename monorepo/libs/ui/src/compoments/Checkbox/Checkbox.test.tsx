import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Checkbox from '.';

const onCheckedChange = jest.fn();

const baseProps = {
  label: 'Label',
  onCheckedChange
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Checkbox {...props} />
    </TamaguiProvider>
  );
};

describe('Checkbox', () => {
  it('should render Checkbox component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Checkbox with checked true', () => {
    const container = setup({checked: true});

    expect(container).toMatchSnapshot();
  });
});
