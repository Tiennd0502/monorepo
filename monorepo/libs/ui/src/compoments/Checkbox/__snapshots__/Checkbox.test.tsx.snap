// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render Checkbox component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "width": 300,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "checked": false,
      }
    }
    checked={false}
    focusVisibleStyle={{}}
    focusable={true}
    id=":r0:"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#424242",
        "borderBottomLeftRadius": 6.25,
        "borderBottomRightRadius": 6.25,
        "borderBottomWidth": 1,
        "borderLeftColor": "#424242",
        "borderLeftWidth": 1,
        "borderRightColor": "#424242",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#424242",
        "borderTopLeftRadius": 6.25,
        "borderTopRightRadius": 6.25,
        "borderTopWidth": 1,
        "flexDirection": "column",
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      }
    }
  />
  <Text
    htmlFor=":r0:"
    id=":r1:"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "color": "#999",
        "display": "flex",
        "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
        "fontSize": 18,
        "lineHeight": 50,
      }
    }
    suppressHighlighting={true}
    userSelect="none"
  >
    Label
  </Text>
</View>
`;

exports[`Checkbox should render Checkbox with checked true 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 8,
      "width": 300,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "checked": false,
      }
    }
    checked={false}
    focusVisibleStyle={{}}
    focusable={true}
    id=":r2:"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242424",
        "borderBottomColor": "#424242",
        "borderBottomLeftRadius": 6.25,
        "borderBottomRightRadius": 6.25,
        "borderBottomWidth": 1,
        "borderLeftColor": "#424242",
        "borderLeftWidth": 1,
        "borderRightColor": "#424242",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#424242",
        "borderTopLeftRadius": 6.25,
        "borderTopRightRadius": 6.25,
        "borderTopWidth": 1,
        "flexDirection": "column",
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      }
    }
  />
  <Text
    htmlFor=":r2:"
    id=":r3:"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "color": "#242424",
        "display": "flex",
        "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
        "fontSize": 18,
        "lineHeight": 50,
      }
    }
    suppressHighlighting={true}
    userSelect="none"
  >
    Label
  </Text>
</View>
`;
